src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/LdapRegistrationDAO.scala

```diff
@@ -93,6 +93,28 @@ class LdapRegistrationDAO(
     }
   }
 
+  //To be used only in the event of the ToS version being bumped
+  override def disableAllHumanIdentities(samRequestContext: SamRequestContext): IO[Unit] = {
+    //The iam.gserviceaccount.com filter is in place to ensure that only human identities are disabled. Service Accounts (both regular SAs and Pet SAs) are
+    // currently exempt for ToS-enforcement, thus, they're ignored when disabling identities
+    val humanIdentityDnsIO = (executeLdap(IO(ldapConnectionPool.search(peopleOu, SearchScope.SUB, "(!(mail=*.iam.gserviceaccount.com))")), "getAllIdentitiesToDisable", samRequestContext) map { results =>
+      results.getSearchEntries.asScala.toList.map { result =>
+        unmarshalUser(result)
+      }
+    }).map { identityResults => identityResults.collect { case Right(user) => subjectDn(user.id) }}
+
+    val enabledIdentityDnsIO = executeLdap(IO(getAttributes(ldapConnectionPool.getEntry(directoryConfig.enabledUsersGroupDn, Attr.member), Attr.member)), "getAllIdentitiesEnabled", samRequestContext).map(_.toList)
+
+    for {
+      humanIdentityDns <- humanIdentityDnsIO //this retrieves all humans registered in the system (looking under the peopleOu)
+      enabledIdentityDns <- enabledIdentityDnsIO //this retrieves all enabled identities in the system (looking in enabledUsersGroupDn)
+      humanIdentityDnsToDisable = humanIdentityDns intersect enabledIdentityDns //intersecting them gives the list of all human identities who are enabled (AKA filtering out humans who are disabled in the system)
```

Comment:
unfortunately this step is necessary because doing an LDAP delete of an attribute value that doesn't exist causes the entire batch to fail. so if there was a user who was already disabled in the system, we need to filter them out first